### PR TITLE
chore(execution): protect `rpc.header` for concurrent access

### PR DESCRIPTION
**Pull request by abi87**
`rpc.header` is access by two different goroutines [rpc.Start()](https://github.com/berachain/beacon-kit/blob/dd024c5b196afe43cf871b5f825a7e371fc4542e/mod/execution/pkg/client/ethclient/rpc/client.go#L88) and [rpc.CallRaw()](https://github.com/berachain/beacon-kit/blob/dd024c5b196afe43cf871b5f825a7e371fc4542e/mod/execution/pkg/client/ethclient/rpc/client.go#L146) respectively.
This PR adds a mutex to safely access header.

## Summary by CodeRabbit

- **New Features**
	- Enhanced thread safety for HTTP header management in RPC requests.

- **Bug Fixes**
	- Resolved potential race conditions when accessing and modifying HTTP headers concurrently.

### CODE REVIEW

File: mod/execution/pkg/client/ethclient/rpc/client.go [hunk 1]
```diff
@@ -46,7 +46,10 @@ type Client struct {
 	// jwtRefershInterval is the interval at which the JWT token should be
 	// refreshed.
 	jwtRefreshInterval time.Duration
-	// header is the HTTP header used for RPC requests.
+
+	// header is the HTTP header used for RPC requests. header is
+	// accessed by multiple goroutines, so it's protected by a mutex.
+	mtx    sync.RWMutex
```
Comment (nidhi-singh02) [inline mod/execution/pkg/client/ethclient/rpc/client.go:52]: nit: Can we rename it to `mu` to keep it consistent across.

**Review by nidhi-singh02 — APPROVED** [review]
lgtm

File: mod/execution/pkg/client/ethclient/rpc/client.go [hunk 2]
```diff
@@ -144,7 +148,10 @@
 	if err != nil {
 		return nil, err
 	}
-	req.Header = rpc.header
+
+	rpc.mu.RLock()
+	req.Header = rpc.header.Clone()
+	rpc.mu.RUnlock()
```
[the remark this answers is not in the document]
Comment (abi87) [reply mod/execution/pkg/client/ethclient/rpc/client.go:154]: I want to lock just enough to clone the header. No need to extend the lock

File: mod/execution/pkg/client/ethclient/rpc/client.go [hunk 3]
```diff
@@ -46,6 +46,10 @@ type Client struct {
 	// jwtRefershInterval is the interval at which the JWT token should be
 	// refreshed.
 	jwtRefreshInterval time.Duration
+
+	// mu protects header for concurrent access.
+	mu sync.RWMutex
```
[the remark this answers is not in the document]
Comment (abi87) [reply mod/execution/pkg/client/ethclient/rpc/client.go:51]: Done

**Review by itsdevbear — APPROVED** [review]

**Pull request merged**